### ISSUE TITLE
Empty guild table in non-english client [Bug] 

### ISSUE THREAD

**Issue by leinstay** [Alumian/CEPGP-Retail#53]
After the recent update of the addon, the guild page with EPGP points became empty. 
![bug1](http://leinstay.oitc.ch/@imgupload/cepgp1.png)

I found and fixed the problem. In non-english clients class in variable will be translated, so something like RAID_CLASS_COLORS[Воин] will break function.

Example in Scrollbar.lua:225
`local colour = RAID_CLASS_COLORS[string.upper(tempTable[i][2])];` 
Usually you use `if class else colour = RAID_CLASS_COLORS["WARRIOR"];`, but not here. 

Please check all similar places. And thank you for developing of this addon!

**Comment by Alumian** [comment]
Hi @leinstay,
Thank you for pointing that out, I actually didn't even think of the impact that would have on non-English clients. I've modified the frame and it will be included in the next hotfix which **should** be today.

**Comment by jpxthu** [comment]
Hi Alumian,

"tempTable[i][2]" will be "战士" in Chinese clients, not "WARRIOR". So "RAID_CLASS_COLORS[string.upper(tempTable[i][2])]" will return nil. In older version (1.11.5), the color will be "RAID_CLASS_COLORS["WARRIOR"]"; but in the later version (1.12.0), the guild and raid table will be empty.

Thanks for your work on this addon!

**Comment by Rguni** [comment]
> 
> 
> Hi @leinstay,
> Thank you for pointing that out, I actually didn't even think of the impact that would have on non-English clients. I've modified the frame and it will be included in the next hotfix which **should** be today.

Hello Alumian!
Thank you for your hard work with addon!
I downloaded last version from Curse Forge. Unfortunately, the problem with empty guild list in russian locale still exists. 

![image](https://user-images.githubusercontent.com/43431755/66681883-594f9b80-ec7c-11e9-9d76-e4d70e82587a.png)

Same goes for raid party list

![image](https://user-images.githubusercontent.com/43431755/66681937-75ebd380-ec7c-11e9-839a-36f45363d248.png)

**Comment by Alumian** [comment]
Hmm... Could you please attach a screenshot of the version frame?

**Comment by Rguni** [comment]
Yes, please.
![image](https://user-images.githubusercontent.com/43431755/66702175-69fc2200-ed0d-11e9-876c-9ca0a9ff9096.png)

**Comment by Alumian** [comment]
Interesting. It looks like it's having a problem with caching the guild roster. I'll check this out.

**Comment by Alumian** [comment]
Are you able to download an addon that can show you lua errors such as stubby or advanced interface options? If you get advanced interface options you'll have to enable lua errors in the addon settings.
Then just log in and copy and paste any error message you see when accessing any of these frames into this thread.

**Comment by Razzari** [comment]
Im not sure. but i have several errors wiht cepgp/
I will write them all

```
Date: 2019-10-13 02:24:46
ID: -4
Error occured in: Global
Count: 25
Message: ..\AddOns\CEPGP\Scrollbar.lua line 227:
   attempt to index local 'colour' (a nil value)
Debug:
   [C]: ?
   CEPGP\Scrollbar.lua:227: CEPGP_UpdateGuildScrollBar()
   CEPGP\Utility.lua:321: CEPGP_populateFrame()
   [string "*:OnClick"]:4:
      [string "*:OnClick"]:1
```

```
ID: -3
Error occured in: Global
Count: 6
Message: ..\AddOns\CEPGP\Scrollbar.lua line 288:
   attempt to index local 'colour' (a nil value)
Debug:
   [C]: ?
   CEPGP\Scrollbar.lua:288: CEPGP_UpdateRaidScrollBar()
   CEPGP\Utility.lua:592: CEPGP_rosterUpdate()
   CEPGP\Core.lua:87: CEPGP_OnEvent()
   [string "*:OnEvent"]:1:
      [string "*:OnEvent"]:1
```

```
ID: -2
Error occured in: Global
Count: 43
Message: ..\AddOns\CEPGP\Scrollbar.lua line 319:
   attempt to index local 'colour' (a nil value)
Debug:
   [C]: ?
   CEPGP\Scrollbar.lua:319: CEPGP_UpdateVersionScrollBar()
   CEPGP\Core.lua:216: ?()
   ..\FrameXML\ChatFrame.lua:4609: ChatEdit_ParseText()
   ..\FrameXML\ChatFrame.lua:4270: ChatEdit_SendText()
   ..\FrameXML\ChatFrame.lua:4306: ChatEdit_OnEnterPressed()
   [string "*:OnEnterPressed"]:1:
      [string "*:OnEnterPressed"]:1
```
```
Date: 2019-10-13 02:30:40
ID: -1
Error occured in: Global
Count: 20
Message: ..\AddOns\CEPGP\Scrollbar.lua line 336:
   attempt to index local 'colour' (a nil value)
Debug:
   [C]: ?
   CEPGP\Scrollbar.lua:336: CEPGP_UpdateVersionScrollBar()
   [string "*:OnClick"]:24:
      [string "*:OnClick"]:1
```
```
ID: 1
Error occured in: Global
Count: 6
Message: ..\AddOns\CEPGP\Scrollbar.lua line 319:
   attempt to index local 'colour' (a nil value)
Debug:
   [C]: ?
   CEPGP\Scrollbar.lua:319: CEPGP_UpdateVersionScrollBar()
   CEPGP\Communications.lua:85: CEPGP_IncAddonMsg()
   CEPGP\Core.lua:139: CEPGP_OnEvent()
   [string "*:OnEvent"]:1:
      [string "*:OnEvent"]:1
```

I hope this will help

**Comment by Alumian** [comment]
Can you please contact me on Discord when you have a moment to troubleshoot this? Alumian#6465

**Comment by leinstay** [comment]
Hi, @Alumian

In recent update nothing changed regarding guild scrollbar, despite your comment in changelog. 
![bug2](http://leinstay.oitc.ch/@imgupload/cepgp3.png)

**Comment by Alumian** [comment]
Good news! Razzari and I have worked out that the problem was caused by the Russian client not recognising the respective class colours for the classes while in Russian. I have written a translation function and will work further on localising the addon but otherwise, this issue has now been resolved and will be included in the next update.
One thing that we did find though, is that I was having difficulties with translating the female equivalents for each class into their male counterparts, but I will try to fix this by localising the addon. The only problem that this causes is the respective players will be in white rather than their class colour.
